include/pybind11/detail/internals.h

```diff
@@ -286,9 +286,11 @@ struct internals {
     internals()
         : static_property_type(make_static_property_type()),
           default_metaclass(make_default_metaclass()) {
+        tstate.set(nullptr);
         PyThreadState *cur_tstate = PyThreadState_Get();
-        tstate = cur_tstate;
-
+        // The PyThreadState that is returned by PyThreadState_Get is not owned by us and may be
+        // destroyed by the owner at some point in the future.  Saving this PyThreadState here can
+        // lead to dangling pointer undefined behavior in complex multi-threaded scenarios.
```

Comment:
This comment only makes sense in the context of this PR, but will be very confusing in the future, because we're not saving "this PyThreadState" here anymore.

WDYT about:
* Deleting this comment entirely.
* Adding a comment above like this:

```
        tstate.set(nullptr); // See PR #5870
```

While at it, could you please also change this line in `pybind11/subinterpreter.h`?

```
    // save this in internals for scoped_gil calls (see also: PR #5870)
    detail::get_internals().tstate = tstate_;
```

(super-easy way to make this context more discoverable for future maintainers)